src/components/Books/api.ts

```diff
@@ -1,22 +1,11 @@
-import IBook from '../../models/IBook';
+import { fetchEndpoint } from '../../business/api';
+import Book from '../../models/Book';
 
-interface IBooksResponse {
-  data: IBook[];
-}
-
-enum ErrorReason {
-  NotOK = 'not ok',
+interface BooksResponse {
+    data: Book[];
 }
 
 const ENDPOINT = 'https://h6jmn6e3vk.execute-api.us-east-1.amazonaws.com/prod/books';
 
-const isError = (response: Response) => !response.ok;
-
-export const fetchBooks = async (): Promise<IBook[]> => {
-  const response = await fetch(ENDPOINT);
-  if (isError(response)) {
-    throw new Error(ErrorReason.NotOK);
-  }
-  const json = await response.json() as IBooksResponse;
-  return json.data;
-};
+export const fetchBooks = async (): Promise<Book[]> =>
+    fetchEndpoint(ENDPOINT, (response: BooksResponse) => response.data);
```

Comment:
I'm not sure where you would wanna stick a util like this, or if you even want to commit to something like this, but since both of your mappers are identical, you might consider a helper function like:

```typescript
export const prop = <V extends object, K extends keyof V>(key: K) => (obj: V): V[K] => obj[key];
```
^ for the type inference to work without you having to do be explicit every time, you would have to update the type signature of `fetchEndpoint` a bit, so feel free to ignore this. You *might* be able to make something like this work, though:

```typescript
fetchEndpoint<BooksResponse>(ENDPOINT, prop('data'));
```

Reply:
Will probably hold off on trying to make this stuff any more generic until I've implemented a few more APIs.